File: 1632 - Quality Assurance/messing_around.js
const readline = require('readline');
const math = require('mathjs');

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

rl.question('Enter an upper bound value > ', (answer) => {
    const upper = answer;

    console.log('Calculating prime numbers...');
    const start = new Date();

    range(2, upper).forEach(function(x) {
        if(math.isPrime(x)) {
            console.log(x + " is prime");
        }
    });
    
    const now = new Date();
    console.log(`Took ${(now - start)/1000} s to test ${upper} values`);

    rl.close();
});

function range(start, end, step, offset) {
  
    var len = (Math.abs(end - start) + ((offset || 0) * 2)) / (step || 1) + 1;
    var direction = start < end ? 1 : -1;
    var startingPoint = start - (direction * (offset || 0));
    var stepSize = direction * (step || 1);
    
    return Array(len).fill(0).map(function(_, index) {
      return startingPoint + (stepSize * index);
    });
    
  }

File: 1632 - Quality Assurance/messing_around.py
import math
import time

def is_prime(n):
    if n % 2 == 0 and n > 2: 
        return False
    return all(n % i for i in range(3, int(math.sqrt(n)) + 1, 2))

upper = int(input("Enter an upper bound value > "))
print ("Calculating prime numbers...")
start = time.time()

values = [x for x in range(2, upper+1) if is_prime(x)]

for val in values:
    print ("{} is prime".format(val))

print("Took {} s to test {} values".format(time.time() - start, upper))
    